src/formatDate.js

```diff
@@ -50,7 +50,69 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const value = date.split(fromFormat[3]);
+  const result = [];
+
+  if (value.join('').length === 6) {
+    for (let i = 0; i < toFormat.length - 1; i++) {
+      if ('YYYY'.includes(toFormat[i])) {
+        for (let k = 0; k < 3; k++) {
+          switch (k) {
+            case toFormat.indexOf('MM'):
+              result.push(value[fromFormat.indexOf('MM')]);
+              break;
+
+            case toFormat.indexOf('DD'):
+              result.push(value[fromFormat.indexOf('DD')]);
+              break;
+
+            default:
+              result.push(value[fromFormat.indexOf('YY')]);
+              break;
+          }
+        }
+      }
+    }
+  } else {
+    for (let i = 0; i < toFormat.length - 1; i++) {
+      if ('YYYY'.includes(toFormat[i])) {
+        for (let k = 0; k < 3; k++) {
+          switch (k) {
+            case toFormat.indexOf('MM'):
+              result.push(value[fromFormat.indexOf('MM')]);
+              break;
+
+            case toFormat.indexOf('DD'):
```

Comment:
we have two almost the same loops, I think we can simplify or combine it, feel free to ask in chat if you have some problems